tests/v2/validation/upgrade/workload_test.go

```diff
@@ -0,0 +1,358 @@
+		require.NoError(u.T(), err)
+		assert.Equal(u.T(), createdIngressForDaemonset.Name, u.names.random["ingressNameForDaemonset"])
+
+		u.T().Logf("Waiting ingress [%v] hostname to be ready", u.names.random["ingressNameForDaemonset"])
+		err = waitUntilIngressHostnameUpdates(client, u.project.ClusterID, u.namespace.Name, u.names.random["ingressNameForDaemonset"])
+		require.NoError(u.T(), err)
+
+		u.T().Logf("Checking if ingress for daemonset with name [%v] is accessible", u.names.random["ingressNameForDaemonset"])
+		ingressForDaemonset, err := ingresses.GetIngressByName(client, u.project.ClusterID, u.namespace.Name, u.names.random["ingressNameForDaemonset"])
+		require.NoError(u.T(), err)
+
+		isIngressForDaemonsetAccessible, err := waitUntilIngressIsAccessible(client, ingressForDaemonset.Spec.Rules[0].Host)
+		require.NoError(u.T(), err)
+		assert.True(u.T(), isIngressForDaemonsetAccessible)
+	}
+
+	if client.Flags.GetValue(environmentflag.Chart) {
+		u.T().Log("Charts tests are enabled")
+
+		u.T().Logf("Checking if the logging chart is installed in cluster [%v]", u.project.ClusterID)
+		loggingChart, err := charts.GetChartStatus(client, u.project.ClusterID, charts.RancherLoggingNamespace, charts.RancherLoggingName)
+		require.NoError(u.T(), err)
+
+		if !loggingChart.IsAlreadyInstalled {
+			clusterName, err := clusters.GetClusterNameByID(client, u.project.ClusterID)
+			require.NoError(u.T(), err)
+			latestLoggingVersion, err := client.Catalog.GetLatestChartVersion(charts.RancherLoggingName)
+			require.NoError(u.T(), err)
+
+			loggingChartInstallOption := &charts.InstallOptions{
+				ClusterName: clusterName,
+				ClusterID:   u.project.ClusterID,
+				Version:     latestLoggingVersion,
+				ProjectID:   u.project.ID,
+			}
+
+			loggingChartFeatureOption := &charts.RancherLoggingOpts{
+				AdditionalLoggingSources: true,
+			}
+
+			u.T().Logf("Installing logging chart with the latest version in cluster [%v] with version [%v]", u.project.ClusterID, latestLoggingVersion)
+			err = charts.InstallRancherLoggingChart(client, loggingChartInstallOption, loggingChartFeatureOption)
+			require.NoError(u.T(), err)
+		}
+	}
+}
+
+func (u *UpgradeWorkloadTestSuite) TestWorkloadPostUpgrade() {
+	subSession := u.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := u.client.WithSession(subSession)
+	require.NoError(u.T(), err)
+
+	defaultListOptions := metav1.ListOptions{
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	}
+
+	namespaceList, err := namespaces.ListNamespaces(client, u.project.ClusterID, defaultListOptions)
+	require.NoError(u.T(), err)
+	doesNamespaceExist := containsItemWithPrefix(namespaceList.Names(), u.names.core["namespaceName"])
+	assert.True(u.T(), doesNamespaceExist)
+
+	if !doesNamespaceExist {
+		u.T().Skipf("Namespace with prefix %s doesn't exists", u.names.core["namespaceName"])
```

Comment:
nit pick

In the kubernetes upgrade version PR this is `doesn't exist` and here `doesn't exists`.

I think, I can be wrong though that it is `doesn't exist`.

Reply:
Thank you, this should be updated as well.